kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/predefined/UpdateProperitesFromDescription.java

```diff
@@ -0,0 +1,157 @@
+import org.cristalise.kernel.lookup.InvalidItemPathException;
+import org.cristalise.kernel.lookup.ItemPath;
+import org.cristalise.kernel.persistency.ClusterType;
+import org.cristalise.kernel.process.Gateway;
+import org.cristalise.kernel.property.Property;
+import org.cristalise.kernel.property.PropertyArrayList;
+import org.cristalise.kernel.property.PropertyDescriptionList;
+import org.cristalise.kernel.property.PropertyUtility;
+import org.cristalise.kernel.utils.Logger;
+
+
+/**
+ * {@value #description}
+ */
+public class UpdateProperitesFromDescription extends PredefinedStep {
+
+    public static final String description = "Updates the Properties of the Item from its description";
+
+    public UpdateProperitesFromDescription() {
+        super();
+    }
+
+    /**
+     * 
+     */
+    protected String runActivityLogic(AgentPath agent, ItemPath item, int transitionID, String requestData, Object locker)
+            throws  InvalidDataException,
+                    InvalidCollectionModification,
+                    ObjectAlreadyExistsException,
+                    ObjectCannotBeUpdated,
+                    ObjectNotFoundException,
+                    PersistencyException,
+                    CannotManageException,
+                    AccessRightsException
+    {
+        String[] inputs = getDataList(requestData);
+
+        String descPath = inputs[0]; //i.e. domainPath of FactoryItem
+        String descVer  = inputs[1];
+        PropertyArrayList initProps = inputs.length == 3 && StringUtils.isNotBlank(inputs[2]) ? unmarshallInitProperties(inputs[2]) : new PropertyArrayList();
+
+        PropertyDescriptionList propDesc = getPropertyDesc(descPath, descVer, locker);
+
+        //Delete or update existing Properties
+        for (String existingPropName: Gateway.getStorage().getClusterContents(item, ClusterType.PROPERTY)) {
+            if (propDesc.definesProperty(existingPropName)) {
+                Property existingProp = PropertyUtility.getProperty(item, existingPropName, locker);
+
+                //TODO: cover the case when the mutable flag was changed
+                if (existingProp.isMutable()) {
+                    //Update existing mutable Property if initial Property list contains it
+                    Property initProp = initProps.get(existingPropName);
+
+                    if (initProp != null) {
+                        existingProp.setValue(initProp.getValue());
+                        Gateway.getStorage().put(item, existingProp, locker);
+                    }
+                }
+                else {
+                    //update existing immutable Property if its default value was changed
+                    String defaultValue = PropertyUtility.getValue(propDesc.list, existingPropName);
+
+                    if (StringUtils.isNotBlank(defaultValue) && !defaultValue.equals(existingProp.getValue())) {
+                        existingProp.setValue(defaultValue);

```

Comment:
so if the existingValue is not the default value, we set it to default? why?

Reply:
If the default value was changed in the definition, the immutable property should have that value.